lib/collection/src/collection/distance_matrix.rs

```diff
@@ -96,7 +96,7 @@ impl From<CollectionSearchMatrixResponse> for SearchMatrixPairsResponse {
             nearests,
         } = response;
 
-        let mut pairs = vec![];
+        let mut pairs = Vec::with_capacity(sample_ids.len() * nearests.len());
```

Comment:
Shouldn't this be `nearest.iter().map(|n| n.len()).sum()` instead?

Nearest is a list of lists, and we add all the items for the sublists.

Or is `sample_ids.len() * nearests.len()` magically the same because it is a square?

Reply:
In the nominal case it should be a square but I think it is better to go with the most precise sizing, thanks :+1:

Reply:
Yeah, though, if it's almost correct we may _just_ have an extra reallocation. And if it should be a square we may benefit from a debug assertion, so we can adjust it once we have a case that isn't square.